attributioncode/validator.go

```diff
@@ -180,6 +181,19 @@ func (v *Validator) Validate(code, sig string) (*Code, error) {
 		rawURLVals: vals,
 	}
 
+	if fromRTAMO(attributionCode.Content) {
+		refererMatch, err := regexp.MatchString(`^https://www.mozilla.org/`, refererHeader)
+		if err != nil {
+			logEntry.Error("Error matching www.mozilla.org regex for RTAMO attribution")
+			return nil, errors.New("Error matching www.mozilla.org regex for RTAMO attribution")
+		}
+
+		if !refererMatch {
+			logEntry.Error("RTAMO attribution does not have https://www.mozilla.org referer header")
+			return nil, errors.New("RTAMO attribution does not have https://www.mozilla.org referer header")
```

Comment:
My understanding is that by returning an error here, [this check](https://github.com/mozilla-services/stubattribution/blob/a41c64d585d16b7a590c021fd00fabf1bc26403b/stubservice/stubhandlers/stubservice.go#L37-L42) will just redirect back to bouncer and give a generic download location. For example, I believe the following `curl` command fails a different check, which results in getting redirected to a non-attributed installer. 
```
$ curl 'https://stubdownloader.services.mozilla.com/?lang=en-US&os=win&product=firefox-stub'
<a href="https://download.mozilla.org/?lang=en-US&amp;os=win&amp;product=firefox-stub">Found</a>.
```

Once this is rolled out, we could try a legitimate request originating from AMO, and just remove the `Referer` header which I believe should have a similar result.